WordPress/Sniffs/WP/EnqueuedResourceParametersSniff.php

```diff
@@ -0,0 +1,155 @@
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		// Check to see if a source ($src) is specified.
+		if ( ! isset( $parameters[2] ) ) {
+			return;
+		}
+
+		/*
+		 * Version Check: Check to make sure the version is set explicitly.
+		 */
+
+		if ( ! isset( $parameters[4] ) || 'null' === $parameters[4]['raw'] ) {
+			$this->phpcsFile->addError(
+				'Resource version not set in call to %s(). This means new versions of the resource will not always be loaded due to browser caching.',
+				$stackPtr,
+				'MissingVersion',
+				array( $matched_content )
+			);
+		} else {
+			// The version argument should have a non-false value.
+			if ( $this->is_falsy( $parameters[4]['start'], $parameters[4]['end'] ) ) {
+				$this->phpcsFile->addError(
+					'Version parameter is not explicitly set for %s; This means that the WordPress core version will be used which is not recommended for plugin or theme development.',
+					$stackPtr,
+					'NoExplicitVersion',
+					array( $matched_content )
+				);
+			}
+		}
+
+		/*
+		 * In footer Check
+		 *
+		 * Check to make sure that $in_footer is set to true.
+		 * It will warn the user to make sure it is intended.
+		 *
+		 * Only wp_register_script and wp_enqueue_script need this check,
+		 * as it is not available to wp_register_style and wp_enqueue_style.
+		 */
+		if ( 'wp_register_script' !== $matched_content && 'wp_enqueue_script' !== $matched_content ) {
+			return;
+		}
+
+		if ( ! isset( $parameters[5] ) ) {
+			// If in footer is not set, throw a warning about the default.
+			$this->phpcsFile->addWarning(
+				'In footer ($in_footer) is not set explicitly %s; ' .
+				'It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.',
+				$stackPtr,
+				'NotInFooter',
+				array( $matched_content )
+			);
+		}
+	}
+
+	/**
+	 * Determine if a range has a falsy value.
+	 *
+	 * @param int $start The position to start looking from.
+	 * @param int $end   The position to stop looking (inclusive).
+	 *
+	 * @return bool True if the parameter is falsy.
+	 */
+	protected function is_falsy( $start, $end ) {
+		$falseTokens   = Tokens::$emptyTokens;
+		$falseTokens[] = T_FALSE;
+
+		// Find anything excluding the false tokens.
+		$hasNonFalse = $this->phpcsFile->findNext( $falseTokens, $start, $end + 1, true );
+		// If no non-false tokens are found, we are good.
+		if ( false === $hasNonFalse ) {
+			return true;
+		}
+
+		// If there is a number, it can be a falsy-value.
```

Comment:
This whole block still needs a little work. Also, the docblock needs to reflect the impreciseness of the method - i.e. `true` means "falsy value found", while `false` can means "no falsy value found or undertermined"

Aside from the regex being invalid - you can't have groups within a character class -, you would also not catch string `'0'` when passed as the parameter value this way. Or `'0' . '.0'` for that matter.

May I suggest an approach which may be slightly more stable ?
* Walk the tokens between `start` and `end`.
* If you encounter an `$emptyToken`, continue.
* If you encounter a token which is not any of in the below list, return `false` (undetermined).
    - `T_NULL`
    - `T_FALSE`
    - `T_TRUE`
    - `T_LNUMBER`
    - `T_DNUMBER`
    - `T_CONSTANT_ENCAPSED_STRING`
    - `T_START_NOWDOC`
    - `T_NOWDOC`
    - `T_END_NOWDOC`
    - `T_OPEN_PARENTHESIS`
    - `T_CLOSE_PARENTHESIS`
    - `T_STRING_CONCAT`
    - `Tokens::$assignmentTokens`
    - `Tokens::$comparisonTokens`
    - `Tokens::$operators`
    - `Tokens::$booleanOperators`
    - `Tokens::$castTokens`
   
    As this array will not change between runs, you can just set the basic token list as a property using the `TOKEN_CODE => true` format and add the token arrays to it from within the `register()` method.
    You can then just do an `isset` in this function to check against the list and exit if something is not on the list.
* If the token is on the list, add it to a temporary variable.
* Once the token walking is done, just do the `false === eval( $tempVar )` as all the tokens I mentioned above will be safe in that context.

For an example of a sniff where I've used a similar approach: https://github.com/wimg/PHPCompatibility/blob/945a8ece200f61e0ea75af5c8242e1eea5ba9b17/PHPCompatibility/Sniffs/PHP/DiscouragedSwitchContinueSniff.php#L174-L186

I believe that using the above approach, all calculations, integers using a different base and the most common results of type juggling etc will be evaluated correctly.

Regarding the `// @codingStandardsIgnoreLine` - the sniff this is copied from is an old sniff from before PHPCS 3.2.0 and as the WPCS minimum PHPCS requirement hasn't gone up yet to 3.2+, existing annotations have not been changed yet.
I agree with @GaryJones though that ignore annotations which are being newly introduced should probably use the new style as the WPCS native CS check in Travis is run on `master` anyway.

Reply:
@moorscode and me discussed this IRL today and I've made the suggested changes in a separate commit (with @moorscode's blessing).